### PR TITLE
1850 Fix YouTube focus bug

### PR DESCRIPTION
**Pull request by Baron-Severin**
### Pull Request checklist

- [x] This PR includes thorough **tests** or an explanation of why it does not
There is no easy way to test this, however we are working on writing Espresso tests for YouTube that will cover this (and other) use cases.
- [X] This PR includes a **CHANGELOG entry** or does not need one
Users have not seen this bug
- [X] The **UI tests** are passing after this PR (`./gradlew connectedSystemDebugAndroidTest`)
- [X] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)

### CODE REVIEW

File: app/src/main/java/org/mozilla/tv/firefox/webrender/FocusedDOMElementCacheInterface.kt [hunk 1]
```diff
@@ -19,7 +19,10 @@ import android.view.View
  *
  * It is important that this value be cached just before the WebView regains focus.  Otherwise
  * edge cases present themselves where a DOM element is cached, a new element is focused by a
- * site, and then we return focus to the earlier (now incorrect) element (see #1830).
+ * site, and then we return focus to the earlier (now incorrect) element (see #1830).  This
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/webrender/FocusedDOMElementCacheInterface.kt]: Nit: Newline break before "This"

File: app/src/system/java/org/mozilla/tv/firefox/ext/EngineView.kt [hunk 2]
```diff
@@ -39,11 +39,12 @@ fun EngineView.setupForApp() {
     // WebView can be null temporarily after clearData(); however, activity.recreate() would
     // instantiate a new WebView instance
     webView?.setOnFocusChangeListener { _, hasFocus ->
-        if (hasFocus) {
-            // See FocusedDOMElementCacheInterface for details on why this happens
-            //
-            // Note that caching of this value is done elsewhere
+        if (!hasFocus) {
```
Comment (liuche) [inline app/src/system/java/org/mozilla/tv/firefox/ext/EngineView.kt]: Hm, does this correlate with the comment changes above? This doesn't look like when the app is going to sleep or backgrounded. Are they unrelated changes?

File: app/src/main/java/org/mozilla/tv/firefox/webrender/FocusedDOMElementCacheInterface.kt (hunk 1 above)
Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/tv/firefox/webrender/FocusedDOMElementCacheInterface.kt]: Are you sure?  It seems to fit with that paragraph, and it's shorter than other lines.

File: app/src/system/java/org/mozilla/tv/firefox/ext/EngineView.kt (hunk 2 above)
Comment (Baron-Severin) [reply app/src/system/java/org/mozilla/tv/firefox/ext/EngineView.kt]: The WV loses focus when the app is backgrounded, so it handles that case as well.  I expanded the comments to make this more clear.

**Comment by liuche** [comment]
@Baron-Severin can you open a PR to the 3.4.1-LAT2 branch too?

**Comment by Baron-Severin** [comment]
@liuche https://github.com/mozilla-mobile/firefox-tv/pull/1860
